transport_nantes/press/views.py

```diff
@@ -165,3 +141,66 @@ class PressMentionDetailView(PermissionRequiredMixin, DetailView):
     model = PressMention
     template_name = "press/press_detail.html"
     permission_required = 'press.press-editor'
+
+
+def fetching_open_graph_data(self, form=False):
+    """
+    This function allow fetch open graph data of an url from
+    a form or from the get data of the url and return
+    a django message.
+    """
+    if form:
+        url = form.cleaned_data['article_link']
+        # Settings the default data
+        form.instance.og_title = form.cleaned_data['article_title']
+        form.instance.og_description = form.cleaned_data['article_summary']
+    else:
+        press_mention = PressMention.objects.get(
+            pk=self.request.GET.get("refresh"))
+        url = press_mention.article_link
+    try:
+        page = requests.get(url)
+    except Exception as e:
+        logger.error(f"Can't acess to the website {e}.")
+        return messages.add_message(
+            self.request, messages.WARNING,
+            'Le liens est incorrect ou le site web est indisponible.')
+    tree = html.fromstring(page.content.decode("utf-8"))
+    og_title = tree.xpath('//meta[@property="og:title"]/@content')
+    og_description = tree.xpath(
+        '//meta[@property="og:description"]/@content')
+    og_image = tree.xpath('//meta[@property="og:image"]/@content')
+    if not og_title or not og_description:
+        logger.error("This website doesn't have all open graph data.")
+        return messages.add_message(
+            self.request, messages.WARNING,
+            'Le site ne contient pas de meta données Open Graph.')
+    else:
+        if form:
+            form.instance.og_title = og_title[0]
+            form.instance.og_description = og_description[0]
+        else:
+            press_mention.og_title = og_title[0]
+            press_mention.og_description = og_description[0]
+    resp = requests.get(og_image[0])
+    if resp.status_code != requests.codes.ok:
+        logger.error("The link of the open graph "
+                     "image may be dead, or doesn't exist.")
+        return messages.add_message(
+            self.request, messages.WARNING,
+            'Le site ne contient pas de meta données Open Graph.')
```

Comment:
Same here, you should probably just state that the image URL is broken, rather than saying there is no data.

Reply:
And, again, the user response to that message should be a mitterrandien "et alors ?".